client/services/utils/local_settings.js

```diff
@@ -0,0 +1,104 @@
+import EventEmitter from 'event-emitter-es6'
+
+const STORAGE_KEY = 'localsettings'
+const DATA_TYPE_KEY = 'dataType'
+const VALUE_KEY = 'value'
+const NAME_KEY = 'name'
+class LocalSettings extends EventEmitter {
+  constructor ($window, $rootScope) {
+    super()
+    Object.assign(this, { $window, $rootScope })
+    // this.STORAGE_KEY = STORAGE_KEY
+    const settingsFromSession = this.getFromLocalStorage()
+    if (settingsFromSession !== undefined) {
+      this.settings = settingsFromSession
+    } else {
+      this.settings = { }
+    }
+  }
+
+  checkSettingValidity (setting) {
+    if (setting.hasOwnProperty(NAME_KEY) && setting.hasOwnProperty(DATA_TYPE_KEY) && setting.hasOwnProperty(VALUE_KEY)) {
+      const dataType = setting[DATA_TYPE_KEY]
+      let valid = false
+      if (dataType === 'number' || dataType === 'boolean' || dataType === 'string') {
+        // eslint-disable-next-line valid-typeof
+        valid = (typeof (setting[VALUE_KEY]) === dataType)
+      }
+      return valid
+    }
+    return false
+  }
+
+  addSettings (namespace, namespaceSettingsArray) {
```

Comment:
This function is used only internally. It should be private (starting with `_` and written at the end of the class).